python/ray/state.py

```diff
@@ -385,6 +385,73 @@ def client_table(self):
 
         return _parse_client_table(self.redis_client)
 
+    def _job_table(self, job_id):
+        """Fetch and parse the job table information for a single job ID.
+
+         Args:
+            job_id: A job ID to get information about.
+         Returns:
+            A dictionary with information about the job ID in question.
+        """
+        # Allow the argument to be either a DriverID or a hex string.
+        if not isinstance(job_id, ray.JobID):
+            assert isinstance(job_id, str)
+            job_id = ray.JobID(hex_to_binary(job_id))
+
+        # Return information about a single job ID.
+        message = self.redis_client.execute_command("RAY.TABLE_LOOKUP",
+            gcs_utils.TablePrefix.Value("JOB"), "", job_id.binary())
+
+        if message is None:
+            return {}
+
+        gcs_entry = gcs_utils.GcsEntry.FromString(message)
+
+        assert len(gcs_entry.entries) > 0
+
+        job_info = {}
+
+        for i in range(len(gcs_entry.entries)):
+            entry = gcs_utils.JobTableData.FromString(gcs_entry.entries[i])
+            assert entry.job_id == job_id.binary()
+            job_info["JobID"] = job_id.hex()
+            job_info["NodeManagerAddress"] = entry.node_manager_address
+            job_info["DriverPid"] = entry.driver_pid
+            if entry.is_dead:
+                job_info["StopTime"] = entry.timestamp
+            else:
+                job_info["StartTime"] = entry.timestamp
+
+        return job_info
+
+    def job_table(self):
+        """Fetch and parse the Redis job table.
+         Returns:
+            Information about the Ray jobs in the cluster,
+            namely a list of dicts with keys:
+            - "JobID" (sha1 identifier for the job),
```

Comment:
```suggestion
            - "JobID" (identifier for the job),
```

It's just random bytes, and will probably be shortened soon.

Reply:
fixed